extension/src/plots/model/collect.ts

```diff
@@ -387,203 +384,96 @@ export const collectSelectedComparisonPlots = ({
-  selectedRevisions
-    .flatMap(revision => {
-      const datapoints = revisionData?.[revision]?.[path] || []
-      return datapoints.map(data => {
-        const obj = data
-        return {
-          ...obj,
-          [key]: mergeFields(fields.map(field => obj[field] as string))
-        }
-      })
-    })
-    .filter(Boolean)
-
-const updateRevisions = (
-  selectedRevisions: string[],
-  domain: string[]
-): string[] => {
-  const revisions: string[] = []
-  for (const revision of selectedRevisions) {
-    for (const entry of domain) {
-      revisions.push(mergeFields([revision, entry]))
-    }
-  }
-  return revisions
-}
-
-const transformRevisionData = (
-  path: string,
-  selectedRevisions: string[],
-  revisionData: RevisionData,
-  isMultiView: boolean,
-  multiSourceEncodingUpdate: { strokeDash: StrokeDashEncoding }
-): { revisions: string[]; datapoints: unknown[] } => {
-  const field = multiSourceEncodingUpdate.strokeDash?.field
-  const isMultiSource = !!field
-  const availableRevisions = selectedRevisions.filter(
-    rev => revisionData[rev]?.[path]
-  )
-  const transformNeeded =
-    isMultiSource && (isMultiView || isConcatenatedField(field))
-
-  if (!transformNeeded) {
-    return {
-      datapoints: selectedRevisions
-        .flatMap(revision => revisionData[revision]?.[path])
-        .filter(Boolean),
-      revisions: availableRevisions
-    }
-  }
-
-  const fields = unmergeConcatenatedFields(field)
-  if (isMultiView) {
-    fields.unshift('rev')
-    return {
-      datapoints: updateDatapoints(
-        path,
-        revisionData,
-        availableRevisions,
-        'rev',
-        fields
-      ),
-      revisions: updateRevisions(
-        availableRevisions,
-        multiSourceEncodingUpdate.strokeDash.scale.domain
-      )
-    }
-  }
-
-  return {
-    datapoints: updateDatapoints(
-      path,
-      revisionData,
-      availableRevisions,
-      field,
-      fields
-    ),
-    revisions: availableRevisions
-  }
-}
-
-const fillTemplate = (
-  template: string,
-  datapoints: unknown[]
-): TopLevelSpec => {
-  return JSON.parse(
-    template.replace('"<DVC_METRIC_DATA>"', JSON.stringify(datapoints))
-  ) as TopLevelSpec
-}
-
 const collectTemplatePlot = (
   acc: TemplatePlotEntry[],
   selectedRevisions: string[],
   path: string,
-  template: string,
+  content: TopLevelSpec,
+  anchorDefinitions: AnchorDefinitions,
   revisionData: RevisionData,
-  nbItemsPerRow: number,
-  height: number,
-  revisionColors: ColorScale | undefined,
-  multiSourceEncoding: MultiSourceEncoding
+  colorScale: ColorScale
 ) => {
-  const isMultiView = isMultiViewPlot(
-    JSON.parse(template) as TopLevelSpec | SpecWithTitles
-  )
-  const multiSourceEncodingUpdate = multiSourceEncoding[path] || {}
-  const { datapoints, revisions } = transformRevisionData(
-    path,
-    selectedRevisions,
-    revisionData,
-    isMultiView,
-    multiSourceEncodingUpdate
-  )
+  const datapoints = selectedRevisions
```

Comment:
[I] Need to make the order make the order they were selected by the user (follow up)